search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -103,34 +93,27 @@ class TemporalEntityHandler(
         } catch (e: BadRequestDataException) {
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).contentType(MediaType.APPLICATION_JSON)
                 .body(BadRequestDataResponse(e.message))
-                .toMono()
         }
 
-        // FIXME this is way too complex, refactor it later
-        return temporalEntityAttributeService.getForEntity(entityId.toUri(), temporalQuery.attrs, contextLink)
-            .switchIfEmpty(Flux.error(ResourceNotFoundException("Entity $entityId was not found")))
-            .flatMap { temporalEntityAttribute ->
-                attributeInstanceService.search(temporalQuery, temporalEntityAttribute)
-                    .map { results ->
-                        Pair(temporalEntityAttribute, results)
-                    }
-            }
-            .collectList()
-            .zipWhen {
-                loadEntityPayload(it[0].first, bearerToken)
-            }
-            .map {
-                val listOfResults = it.t1.map {
-                    it.second
-                }
-                temporalEntityAttributeService.injectTemporalValues(it.t2, listOfResults, withTemporalValues)
-            }
-            .map {
-                JsonLdUtils.filterCompactedEntityOnAttributes(it.compact(), temporalQuery.attrs)
-            }
-            .map {
-                ResponseEntity.status(HttpStatus.OK).body(serializeObject(it))
-            }
+        val temporalEntityAttribute = temporalEntityAttributeService.getForEntity(
+            entityId.toUri(),
+            temporalQuery.attrs,
+            contextLink
+        ).awaitFirstOrNull() ?: throw ResourceNotFoundException("Entity $entityId was not found")
```

Comment:
I think you can use the ENTITY_NOT_FOUND_MESSAGE in ApiResponses.kt in shared module.

Reply:
OK , I am adding a function in ApiResponses.kt for the message including entityId

Reply:
you can use it in TemporalEntityHandlerTests too 
`it should return a 404 if temporal entity attribute does not exist`

Reply:
OK